src/server.c

```diff
@@ -838,6 +840,13 @@ static void cxtr_try_open(void *context)
         return;
     }
 
+    // Set the sasl user name and password on the proton connection object. This has to be done before the call to qdpn_connector_transport() which
+    // binds the transport to the connection
+    if(config->sasl_username && config->sasl_password) {
+        pn_connection_set_user(ctx->pn_conn, config->sasl_username);
+        pn_connection_set_password(ctx->pn_conn, config->sasl_password);
```

Comment:
I think you should treat the username and password independently.  You shouldn't assume that there will always be both present.